### ISSUE TITLE
tfjs-node installation on windows

### ISSUE THREAD

**Issue by youssefMo99** [tensorflow/tfjs#8064]
when running this command: "node-gyp configure --verbose" in "node_modules\@tensorflow\tfjs-node",
i get the following error:
gyp: Undefined variable module_name in binding.gyp while trying to load binding.gyp
gyp ERR! configure error
gyp ERR! stack Error: `gyp` failed with exit code: 1
gyp ERR! stack at ChildProcess.<anonymous> (C:\Users\ASUS\AppData\Roaming\nvm\v20.9.0\node_modules\node-gyp\lib\configure.js:270:18)
gyp ERR! stack at ChildProcess.emit (node:events:514:28)
gyp ERR! stack at ChildProcess._handle.onexit (node:internal/child_process:294:12)
gyp ERR! System Windows_NT 10.0.22621
gyp ERR! command "C:\\Program Files\\nodejs\\node.exe" "C:\\Program Files\\nodejs\\node_modules\\node-gyp\\bin\\node-gyp.js" "configure" "--verbose"        
gyp ERR! cwd C:\Users\ASUS\Desktop\tensorFlow\node_modules\@tensorflow\tfjs-node
gyp ERR! node -v v20.9.0
gyp ERR! node-gyp -v v10.0.0
gyp ERR! not ok

any thoughts on how to fix this?

**Comment by gaikwadrahul8** [comment]
Hi, @youssefMo99 

Thank you for bringing this issue to our attention, I see you're using the `Node.js v20.9.0` version so could you please give it try by downgrading to `Node.js v18.16.1` with `python 3.8` or `python 3.9` and try with `npm install --save-exact @tensorflow/tfjs-node@3.18.0 `

I believe you're referring this [official documentation ](https://github.com/tensorflow/tfjs/tree/master/tfjs-node), [TensorFlow.js Node.js bindings Windows troubleshooting.](https://github.com/tensorflow/tfjs/blob/master/tfjs-node/WINDOWS_TROUBLESHOOTING.md#tensorflowjs-nodejs-bindings-windows-troubleshooting), [node-gyp on Windows](https://github.com/nodejs/node-gyp#on-windows)

In general you'll have to follow below steps to install `@tensorflow/tfjs-node` on `Windows` system:

- Edit the path user or system variable, ensure that the path to python is added to the path variable.
- Install Visual Studio community edition with desktop development with C++ [Link](https://visualstudio.microsoft.com/vs/community/)
- Install visual studio build tools with the help of Visual Studio Installer
- Ensure that node-gyp is available: `npm install -g node-gyp`
- Install tfjs-node: `npm install --save-exact @tensorflow/tfjs-node@3.18.0`

After following above instructions If issue still persists please let us know with error log after running `node-gyp configure --verbose` command. Thank you.

**Comment by gaikwadrahul8** [comment]
Hi, @KristianBic 

You're welcome and good to hear that you're able to install `tfjs-node `on your windows machine. as far I know that is an optional warning message from TensorFlow informing you that your CPU supports certain instructions `AVX2` and `AVX`, but the TensorFlow binary was not compiled to use them. 

TensorFlow being used has been built using default options. This can happen if you have install TensorFlow from pip or built from source without enabling optimization flags

This means that TensorFlow may not be able to take full advantage of your `CPU` power. If you want TensorFlow to use these instructions `AVX2` and `AVX`, you will need to rebuild it with the appropriate compiler flags, If you are using TensorFlow for computationally intensive tasks, such as training a complex neural network, it may be worth it to rebuild TensorFlow with the appropriate compiler flags to take advantage of these instructions or else you can ignore that warning message.

If you have a `GPU`, you shouldn't care about `AVX2` and `AVX` instructions support, because most expensive ops will be dispatched on a `GPU` device, you can ignore that warning message by using below code

```
import os
os.environ[‘TF_CPP_MIN_LOG_LEVEL’] = ‘2’
```

I hope I answered your questions and if you're looking to rebuild tensorflow with the appropriate compiler flags then this [article](https://iq.opengenus.org/tensorflow-binary-was-not-compiled-to-use/) may help you to use these instructions `AVX2` and `AVX` to speed up your mathematical calculations. Thank you.

**Comment by gaikwadrahul8** [comment]
Hi, @youssefMo99 

I'm able to install tfjs-node on my `windows 10` system by following the instructions mentioned in my https://github.com/tensorflow/tfjs/issues/8081#issuecomment-1827814656 with these Node.js versions `Node.js v19.9.0 `and `Node.js v18.16.1` and at the moment it's not working as expected with Node.js version `v20.10.0` and our relevant team is working on this issue to fix it

You should able to install `tfjs-node` on your windows system with Node.js versions `Node.js v19.9.0` `Node.js v18.16.1` so now I'm closing this issue, if you face any issue while installing `tfjs-node` after following above mentioned instructions with mentioned Node.js versions please feel free to create new issue with your new error log and would be glad to help you further

Thank you for your understanding and patience.

1. With `Node.js v19.9.0` :

```
PS C:\Users\gaikwadrahul\Desktop\tfjs-node-testing-for-doc\tfjs-node19-py-3.10> python
Python 3.10.9 (tags/v3.10.9:1dd9be6, Dec  6 2022, 20:01:21) [MSC v.1934 64 bit (AMD64)] on win32
Type "help", "copyright", "credits" or "license" for more information.
>>> quit()
PS C:\Users\gaikwadrahul\Desktop\tfjs-node-testing-for-doc\tfjs-node19-py-3.10> node -v
v19.9.0
PS C:\Users\gaikwadrahul\Desktop\tfjs-node-testing-for-doc\tfjs-node19-py-3.10> npm -v
9.6.3
PS C:\Users\gaikwadrahul\Desktop\tfjs-node-testing-for-doc\tfjs-node19-py-3.10> pyenv which python
C:\Users\gaikwadrahul\.pyenv\pyenv-win\versions\3.10.9\python.exe
PS C:\Users\gaikwadrahul\Desktop\tfjs-node-testing-for-doc\tfjs-node19-py-3.10> npm install -g node-gyp --python="C:\Users\gaikwadrahul\.pyenv\pyenv-win\versions\3.10.9\python.exe”   

changed 99 packages in 6s
PS C:\Users\gaikwadrahul\Desktop\tfjs-node-testing-for-doc\tfjs-node19-py-3.10> npm install --python="C:\Users\gaikwadrahul\.pyenv\pyenv-win\versions\3.10.9\python.exe" @tensorflow/tfjs-node

changed 10 packages, and audited 120 packages in 51s

9 packages are looking for funding
  run `npm fund` for details

found 0 vulnerabilities
PS C:\Users\gaikwadrahul\Desktop\tfjs-node-testing-for-doc\tfjs-node19-py-3.10> 
```

2. with` Node.js v18.16.1` :

```
PS C:\Users\gaikwadrahul\Desktop\tfjs-node-testing-for-doc\tfjs-node-3.10> python
Python 3.10.9 (tags/v3.10.9:1dd9be6, Dec  6 2022, 20:01:21) [MSC v.1934 64 bit (AMD64)] on win32
Type "help", "copyright", "credits" or "license" for more information.
>>> quit()
PS C:\Users\gaikwadrahul\Desktop\tfjs-node-testing-for-doc\tfjs-node-3.10> node --version
v18.16.1
PS C:\Users\gaikwadrahul\Desktop\tfjs-node-testing-for-doc\tfjs-node-3.10> npm --version
9.8.0
PS C:\Users\gaikwadrahul\Desktop\tfjs-node-testing-for-doc\tfjs-node-3.10> pyenv which python
C:\Users\gaikwadrahul\.pyenv\pyenv-win\versions\3.10.9\python.exe
PS C:\Users\gaikwadrahul\Desktop\tfjs-node-testing-for-doc\tfjs-node-3.10> npm install -g node-gyp --python="C:\Users\gaikwadrahul\.pyenv\pyenv-win\versions\3.10.9\python.exe"   

changed 99 packages in 12s

14 packages are looking for funding
  run `npm fund` for details
PS C:\Users\gaikwadrahul\Desktop\tfjs-node-testing-for-doc\tfjs-node-3.10> npm install --python="C:\Users\gaikwadrahul\.pyenv\pyenv-win\versions\3.10.9\python.exe" @tensorflow/tfjs-node

added 121 packages, and audited 122 packages in 3m

9 packages are looking for funding
  run `npm fund` for details

found 0 vulnerabilities
PS C:\Users\gaikwadrahul\Desktop\tfjs-node-testing-for-doc\tfjs-node-3.10> node .\index.js
2023-11-30 15:04:31.614170: I tensorflow/core/platform/cpu_feature_guard.cc:193] This TensorFlow binary is optimized with oneAPI Deep Neural Network Library (oneDNN) to use the following CPU instructions in performance-critical operations:  AVX2
To enable them in other operations, rebuild TensorFlow with the appropriate compiler flags.
Epoch 1 / 3
eta=0.0 ==================================================================================> 
29686ms 2969us/step - MAE=0.332 loss=0.168 val_MAE=0.293 val_loss=0.126 
Epoch 2 / 3
eta=0.0 ==================================================================================>
9458ms 946us/step - MAE=0.279 loss=0.113 val_MAE=0.271 val_loss=0.106 
Epoch 3 / 3
eta=0.0 ==================================================================================> 
11791ms 1179us/step - MAE=0.258 loss=0.0940 val_MAE=0.275 val_loss=0.110 
PS C:\Users\gaikwadrahul\Desktop\tfjs-node-testing-for-doc\tfjs-node-3.10> 
```

**Comment by Fatimaarshad10** [comment]
Thank you. It resolved my issue.
